src/gui/menu_filesystem.cpp

```diff
@@ -111,6 +114,9 @@ FileSystemMenu::refresh_items()
     if (in_basedir && m_item_processor)
       m_item_processor(menu_item);
 
+    const std::string help_text = m_generate_help_text_for_file ? m_generate_help_text_for_file(FileSystem::join(m_directory, item)) : "";
+    menu_item.set_help(help_text);
+
```

Comment:
To achieve this, an item processor in `m_item_processor` can be provided.

Reply:
I agree! I refactored the code to use `m_item_processor`.